enterprise/frontend/src/metabase-enterprise/troubleshooting/utils.ts

```diff
@@ -0,0 +1,55 @@
+import { t } from "ttag";
+import _ from "underscore";
+
+import type { CardError } from "metabase-types/api";
+
+const ERROR_DICTIONARY = {
+  "inactive-field": {
+    entity: t`Field`,
+    messageProp: "field" as const,
+    problem: t`is inactive`,
+  },
+  "unknown-field": {
+    entity: t`Field`,
+    messageProp: "field" as const,
+    problem: t`is unknown`,
+  },
+  "inactive-table": {
+    entity: t`Table`,
+    messageProp: "table" as const,
+    problem: t`is inactive`,
+  },
+  "unknown-table": {
+    entity: t`Table`,
+    messageProp: "table" as const,
+    problem: t`is unknown`,
+  },
+};
+
+export const formatErrorString = (errors: CardError[]) => {
+  const messages: string[] = [];
+
+  const errorsByType = _.groupBy(errors, "type");
+
+  const errorTypes = Object.keys(
+    ERROR_DICTIONARY,
+  ) as (keyof typeof ERROR_DICTIONARY)[];
```

Comment:
two minor nits
- this can be moved out of this function so it's only done once
- i think using _.keys will do the same without having to do the typecasting